resources/schemas/dbscripts/sqlserver/snd-20.000-20.001.sql

```diff
@@ -0,0 +1,73 @@
+/*
+    Changes to improve snd.Events CRUD performance and add default QcState
+*/
+
+-- delete the trigger if it exists
+IF (OBJECT_ID(N'snd.ti_after_Events') IS NOT NULL)
+   BEGIN
+       DROP TRIGGER snd.ti_after_Events;
+   END;
+GO
+-- if QcState is null, set it to 'Completed'
+CREATE TRIGGER snd.ti_after_Events ON snd.Events FOR INSERT AS
+BEGIN
+    SET NOCOUNT ON;
+    UPDATE snd.Events
+    SET QcState = (SELECT TOP(1) q.rowId FROM core.Qcstate AS q WHERE q.Label = 'Completed' ORDER BY q.rowId)
+    FROM inserted AS i
+             INNER JOIN snd.Events AS e ON i.EventId = e.EventId
+    WHERE i.QcState IS NULL
+END
+go
+
+-- move Events table cluster index to AnimalId, Date
+--
+-- need to drop foreign key constraints that reference the cluster index
+EXEC core.fn_dropifexists 'EventNotes', 'snd', 'CONSTRAINT', 'FK_SND_EVENTNOTES_EVENTID'
+EXEC core.fn_dropifexists 'EventData', 'snd', 'CONSTRAINT', 'FK_SND_EVENTDATA_EVENTID'
+EXEC core.fn_dropifexists 'EventsCache', 'snd', 'CONSTRAINT', 'FK_EventsCache_EventId'
+-- drop the snd.Events PK constraint (clustered index)
+EXEC core.fn_dropifexists 'Events', 'snd', 'CONSTRAINT', 'PK_SND_EVENTS'
```

Comment:
Uniqueness of these is mostly managed by Java, but we should probably put unique constraints on these Ids since not PK anymore.

Reply:
Lines 32-46 add a unique index on Events. EventId

Reply:
The FK constraints are recreated on lines 49-73

Reply:
Ah got it, I missed which constraints we were dropping there. Looks good.